charts/nd-common/templates/_istio.tpl

```diff
@@ -79,6 +79,25 @@ proxy.istio.io/overrides: >-
   } 
 {{- end }}
 
+{{/*
+
+Prepare the set of resource annotations for the istio-sidecar.
+Fallbacking to the current implemented default resources:
+    Limits:
+      cpu:                2
+      memory:             1Gi
+    Requests:
+      cpu:                100m
+      memory:             128Mi
+
+*/}}
+{{- if or (or .Values.istio.proxyCPU .Values.istio.proxyMemory) (or .Values.istio.proxyCPULimit .Values.istio.proxyMemoryLimit) }}
```

Comment:
I don't love that we are copy and pasting the defaults here. An alternative approach is to use `{{- with ...` syntax for each setting. Additionally, I'd love to use standard Kubernetes "resources" syntax. Eg:

```yaml
myApp:
  istio:
    resources:
      requests:
        cpu: 2
        memory: 1Gi
```

Then the template can look something like this:

```
{{- with .Values.istio }}
{{- with .resources }}

{{- with .requests }}
{{- with .cpu }}
sidecar.istio.io/proxyCPU: {{ . | quote }}
{{- end }}
{{- with .memory }}
sidecar.istio.io/proxyMemory: {{ . | quote }}
{{- end }}
{{- end }}

{{- with .limits }}
{{- with .cpu }}
sidecar.istio.io/proxyCPULimit: {{ . | quote }}
{{- end }}
{{- with .memory }}
sidecar.istio.io/proxyMemoryLimit: {{ . | quote }}
{{- end }}
{{- end }}

{{- end }}
{{- end }}
```

## Example - 3 out of 4 settings set:
```diff
+ istio:
+   resources:
+     requests:
+       cpu: 10m
+       memory: 10Mi
+     limits:
+       memory: 60Mi
```

```diff
+    helm.sh/chart: simple-app-1.6.1
     app.kubernetes.io/version: "latest"
     app.kubernetes.io/managed-by: Helm
     tags.datadoghq.com/service: "simple-app"
@@ -242,6 +242,9 @@
               }
             ]
           }
+        sidecar.istio.io/proxyCPU: "10m"
+        sidecar.istio.io/proxyMemory: "10Mi"
+        sidecar.istio.io/proxyMemoryLimit: "60Mi"
         ad.datadoghq.com/simple-app.check_names: '["prometheus"]'
         ad.datadoghq.com/simple-app.init_configs: '[{}]'
         ad.datadoghq.com/simple-app.instances: |-
@@ -261,7 +264,7 @@
```

## Example: Only limits set
```diff
+ istio:
+   resources:
+     requests:
+     limits:
+       memory: 60Mi
```

```diff
   labels:
-    helm.sh/chart: simple-app-1.6.0
+    helm.sh/chart: simple-app-1.6.1
     app.kubernetes.io/version: "latest"
     app.kubernetes.io/managed-by: Helm
     tags.datadoghq.com/service: "simple-app"
@@ -242,6 +242,8 @@
               }
             ]
           }
+        sidecar.istio.io/proxyCPULimit: "400m"
+        sidecar.istio.io/proxyMemoryLimit: "60Mi"
         ad.datadoghq.com/simple-app.check_names: '["prometheus"]'
         ad.datadoghq.com/simple-app.init_configs: '[{}]'
         ad.datadoghq.com/simple-app.instances: |-
@@ -261,7 +263,7 @@
             }
```

Reply:
I'm OK with changing the format to match the one from k8s, the with thought will not work.
The reason is that it won't generate the annotation if the value is not set and that's necessary to avoid the issue we had.

Reply:
Hmm. Are you saying that we need _all four annotations_ or they won't be used at all? I am under the impression that each annotation can be used individually.

Reply:
Yes, the issue was that just changing the cpu requests, cause the memory limits to be set to the memory requests - which cause the sidecar to OOM :-(

Reply:
If I set the default values in the value file, I could use your with logic though, would you prefer that?

Reply:
You can also look at the current change, which is another approach while leaving the values files empty if you prefer.

Reply:
For now I'll leave it like that, we can always change it(the end result is the same).